### ISSUE TITLE
Lesson 8.11 RRT Exercise Errors

### ISSUE THREAD

**Issue by tjphoton** [udacity/fcnd-issue-reports#170]
Lesson 8.11 RRT Exercise, in RRT-Solution.ipynb,
1. sklearn package is not included in the fcnd starter environment. 
need to do the `# !{sys.executable} -m pip install sklearn` trick to have it installed.

2. In the obstacle visualization code cell `plt.imshow(grid, cmap='Greys', origin='upper')`, `upper` should be changed to `lower` to be consistent with the final visualization.

3. In the code cell 11, it only check whether the new state is located on an obstacle, but did not check whether the new state is out of grid boundary, nor check whether the new edge to be added will pass through obstacles.

4. Again, obstacle boundary index issue.  When marked obstacles `grid[30:40, 60:82] = 1`,  there are 10 grid cell index marked in the x direction: [30, 31, 32, 33, 34, 35, 36, 37, 38, 39].  In the visualization with `plt.imshow(grid)`, region between 29.5 and 29.5 will be colored black. This caused feasible states with x axis between 29.5 and 30 is misleadingly in obstacle region.

**Comment by tjphoton** [comment]
![RRT_edge_through.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a69fda3bc91ae00128ab4c5/125516c66e82c728ace21e0d46daa6ae1c8787ae8ad6ab4bb652e59321046b313916ac217586f76ae5002e00595e4da4405a5646f5ac812aa6b63e77d90f5de5d26f4d68d47f40f2ecdd42e64bef62994597b4bd804dc5ea3c26e58a8d46391b05109bb2.png)
![RRT_obstacle_index.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5a69fda3bc91ae00128ab4c5/125516c66e82c728ace21e0d46daa6ae1c8d81ba9be8bc4fa162f99a2d433f316546b3397f8fe62bab0021435a0e48ad42505e46f5fb807eaeb73c70885b58be87655c76d42103eee9dc4be05ce36ade0d97bcae884dcae93827e6888d4e3c17021c81f94f78.png)

**Comment by tjphoton** [comment]
Correction: 10 grid cell index marked in the *y* direction

**Comment by domluna** [comment]
@tjphoton 

1) scikit-learn has been added https://github.com/udacity/FCND-Term1-Starter-Kit/blob/master/environment.yml#L14
2) done 👍 
3/4) I don't think there's much we can do about how matplotlib plots things. We can use `int(np.round(x))` instead of `int(x)`. This would be more in line with how matplotlib plots things. However, since we're not checking if there's an obstacle between 2 points there's still the possibly a line will be drawn through the corner of an obstacle.
